lib/services/event_message_handler.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:async';
+
+import 'package:firebase_database/firebase_database.dart';
+import 'package:regatta_buddy/models/message.dart';
+
+class EventMessageHandler {
+  final String eventId;
+  final String? teamId;
+  void Function(Message)? onStartEventMessage;
+  StreamSubscription<DatabaseEvent>? messageStream;
+
+  EventMessageHandler(
+      {required this.eventId, this.teamId, this.onStartEventMessage});
+
+  bool _isReferee() {
+    return teamId == null;
+  }
+
+  void start() {
+    final databaseReference = FirebaseDatabase.instance.ref();
+    DatabaseReference messRef =
+        databaseReference.child('messages').child(eventId);
+
+    messageStream = messRef.onChildAdded.listen((event) {
+      final messageData = Map<String, String>.from(
+          event.snapshot.value as Map<Object?, Object?>);
+      final message = Message.fromJson(messageData);
+      print(message);
+
+      if (message.receiverType == MessageReceiverType.all ||
```

Comment:
that's a long if statement haha

Reply:
yep, I will try to make this better in next pr :D